packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/ClaimButton/ClaimButton.test.ts

```diff
@@ -0,0 +1,179 @@
+import { describe, expect, it } from "vitest";
+import { ANVIL_CHAIN } from "~test/chains.js";
+import { TEST_CONTRACT_URI } from "~test/ipfs-uris.js";
+import { TEST_CLIENT } from "~test/test-clients.js";
+import { DOODLES_CONTRACT, USDT_CONTRACT } from "~test/test-contracts.js";
+import { TEST_ACCOUNT_A } from "~test/test-wallets.js";
+import { getContract } from "../../../../../../contract/contract.js";
+import { deployERC20Contract } from "../../../../../../extensions/prebuilts/deploy-erc20.js";
+import { deployERC721Contract } from "../../../../../../extensions/prebuilts/deploy-erc721.js";
+import { deployERC1155Contract } from "../../../../../../extensions/prebuilts/deploy-erc1155.js";
+import {
+  getERC20ClaimTo,
+  getERC721ClaimTo,
+  getERC1155ClaimTo,
+} from "./index.js";
+
+const account = TEST_ACCOUNT_A;
+const client = TEST_CLIENT;
+const chain = ANVIL_CHAIN;
+
+/**
+ * The ClaimButton uses TransactionButton under the hood
+ * So all we need to test is whether it gives the correct PreparedTransaction to the TransactionButton
+ */
+describe("ClaimButton", () => {
+  // ERC721
+  it("should throw an error if not an ERC721 contract", async () => {
+    const contract = USDT_CONTRACT;
+    await expect(() =>
+      getERC721ClaimTo({
+        contract,
+        account,
+        claimParams: { type: "ERC721", quantity: 1n },
+      }),
+    ).rejects.toThrowError("Not an ERC721 contract");
+  });
+
+  it("should throw an error if is an ERC721 but not an NFT Drop contract", async () => {
+    const contract = DOODLES_CONTRACT;
+    await expect(() =>
+      getERC721ClaimTo({
+        contract,
+        account,
+        claimParams: { type: "ERC721", quantity: 1n },
+      }),
+    ).rejects.toThrowError("Not a valid NFT Drop (ERC721) contract");
+  });
+
+  it("should work for an NFT Drop contract", async () => {
+    const address = await deployERC721Contract({
+      client,
+      chain,
+      account,
+      type: "DropERC721",
+      params: {
+        name: "",
+        contractURI: TEST_CONTRACT_URI,
+      },
+    });
+    const contract = getContract({ address, client, chain });
+    const result = await getERC721ClaimTo({
+      contract,
+      account,
+      claimParams: { type: "ERC721", quantity: 1n },
+    });
+    expect(result).toBeDefined();
```

Comment:
Is there anything more specific we can check for?

Reply:
I couldn't use encode() or await result.data() because it throws error if you don't have an active claim condition set up + no nft minted.

Anyway I have updated the test to be more strict